tests/scripts/all.sh

```diff
@@ -3448,6 +3448,24 @@ component_test_malloc_0_null () {
     tests/ssl-opt.sh -e 'proxy'
 }
 
+support_test_aesni () {
+    # require an x64_64 target
+    gcc -v 2>&1 | grep Target | grep -q x86_64
```

Comment:
This is not enough to test that the target supports AESNI. We could be running the tests on an x86_64 platform without AES acceleration.

Reply:
I think that's OK, the target will use run-time selection and should run correctly regardless.

Reply:
Addressed via a comment.

Reply:
I have separated the checks for "can build" and "can run" (we require both).

Reply:
As you remarked offline, the name of the `support_xxx` function has to match the `component_xxx` function: it means “can we run the xxx component?”. So all that's needed here is a comment that explains that our AESNI _tests_ require an x86_64 platform, and we assume that all x86_64 platforms involved in our CI support AESNI.

At the moment we don't do cross-compiling for x86 on the CI so either `gcc | grep Target` or `uname` will do (or the mostly equivalent `HOSTTYPE` — but when we start mixing architectures that can be different since `HOSTYTPE` is specifically the architecture that bash was built for). If we do cross-compiling, we might be executing the tests through transparent cross-architecture execution via Linux's binmft_misc mechanism and qemu, so `gcc | grep Target` alone would be the correct test.